tests/ValidatorAttributesTest.php

```diff
@@ -0,0 +1,139 @@
+use MayMeow\ExcelImporter\Attributes\MaxLength;
+use MayMeow\ExcelImporter\Attributes\Min;
+use MayMeow\ExcelImporter\Attributes\MinLength;
+use MayMeow\ExcelImporter\Attributes\NotEmpty;
+use MayMeow\ExcelImporter\Attributes\Regex;
+use MayMeow\ExcelImporter\Attributes\Required;
+use MayMeow\ExcelImporter\Attributes\Url;
+use PHPUnit\Framework\TestCase;
+
+class ValidatorAttributesTest extends TestCase
+{
+    /** @test */
+    public function testRequiredAttribute()
+    {
+        $required = new Required();
+        $this->assertTrue($required->validate(''));
+        $this->assertTrue($required->validate('value'));
+        $this->assertTrue($required->validate(0));
+        $this->assertFalse($required->validate(null));
+    }
+
+    /** @test */
+    public function testNotEmptyAttribute()
+    {
+        $notEmpty = new NotEmpty();
+        $this->assertFalse($notEmpty->validate(''));
+        $this->assertTrue($notEmpty->validate('value'));
+        $this->assertTrue($notEmpty->validate(0));
+        $this->assertFalse($notEmpty->validate(null));
+    }
+
+    /** @test */
+    public function testMaxLengthAttribute()
+    {
+        $maxLength = new MaxLength(5);
+        $this->assertTrue($maxLength->validate('12345'));
+        $this->assertTrue($maxLength->validate('123'));
+        $this->assertFalse($maxLength->validate('123456'));
+        $this->assertFalse($maxLength->validate(123456));
+    }
+
+    /** @test */
+    public function testMinLengthAttribute()
+    {
+        $minLength = new MinLength(5);
+        $this->assertTrue($minLength->validate('12345'));
+        $this->assertTrue($minLength->validate('123456'));
+        $this->assertFalse($minLength->validate('123'));
+        $this->assertFalse($minLength->validate(123));
+    }
+
+    /** @test */
+    public function testMinAttribute()
+    {
+        $min = new Min(5);
+        $this->assertTrue($min->validate(5));
+        $this->assertTrue($min->validate(6));
+        $this->assertFalse($min->validate(4));
+        $this->assertFalse($min->validate('4'));
+        $this->assertFalse($min->validate('abc'));
+    }
+
+    /** @test */
+    public function testMaxAttribute()
+    {
+        $max = new Max(5);
+        $this->assertTrue($max->validate(5));
+        $this->assertTrue($max->validate(4));
+        $this->assertFalse($max->validate(6));
+        $this->assertFalse($max->validate('6'));
+        $this->assertFalse($max->validate('abc'));
+    }
+
+    /** @test */
+    public function testBetweenAttribute()
+    {
+        $between = new Between(5, 10);
+        $this->assertTrue($between->validate(5));
+        $this->assertTrue($between->validate(7));
+        $this->assertTrue($between->validate(10));
+        $this->assertFalse($between->validate(4));
+        $this->assertFalse($between->validate(11));
+        $this->assertFalse($between->validate('7'));
+        $this->assertFalse($between->validate('abc'));
```

Comment:
Yours testing passing because you checking if the value is string or not.

https://github.com/MayMeow/excel-importer/blob/6bb7661f2020f61422be9084a2fb5c81f5c0be63/src/Attributes/Between.php#L20-L22

but you should test for string lenght instead.

Reply:
Can you also add docblock for it?